tests/tests/kernel/datatypes/ezimage/ezimagealiashandler_regression.php

```diff
@@ -88,6 +82,38 @@ public function testIssue15155()
         array_map( 'unlink', $toDelete );
         $image->purge();
     }
+
+    /**
+     * @return array
+     */
+    public function providerForTestTrimToFileSystemFileName()
+    {
+        return array(
+            array(
+                "This is a Very Long Name isn't it?",
+                20,
+                "This is a Very Long "
+            ),
+            array(
+                "私は簡単にパブリッシュの記事で使用することができるようなも",
+                20,
+                "私は簡単にパ"
+            ),
+            array(
+                "私aはb簡c単dにeパfブgリhッシュの記事で使用することができるようなも",
+                20,
+                "私aはb簡c単dにe"
+            )
+        );
+    }
+
+    /**
+     * @dataProvider providerForTestTrimToFileSystemFileName
+     */
+    public function testTrimToFileSystemFileName( $longName, $limit, $expects )
+    {
+        $this->assertEquals( $expects, eZImageAliasHandler::trimToFileSystemFileName( $longName, $limit ) );
```

Comment:
Actually since we are spliting UTF-8 chars, triming to 20 bytes a string can provide a 18 bytes lenght string.